tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -2411,6 +2418,36 @@ private String handleTimeZoneName(XPathParts parts, String value) {
         return result;
     }
 
+    private String generateAltPattern(String dateTimeValue) {
```

Comment:
Rereading the ticket, I see "the caller can specify the width of _various fields_." Right now, I'm only accounting for different widths of hours (I thought, at first, that only hours were relevant). Should I be checking every variable field? In that case, couldn't the number of possible examples be very high (e.g. if all of h,m,s can have their width changed, would 8 examples be desired)?

Reply:
For time, we only allow the first to be padded or not, eg:

9:08:07 or 09:08:07, but not 9:8:7

Reply:
Add comment that this function could be generalizes for fields other than h/H. (If you had more time, you could do it, but the hourglass has run out....)